Newsletter fixes

Oldiesmann opened the pull request:
This addresses a couple of issues:
- Always include a link for {$member.link} even when not sending via HTML (fixes #4116)
- Flawed logic in the code caused the newsletter to not get sent in most cases

Oldiesmann committed:
Always include a link for {$member.link} in newsletters even when not sending via HTML (fixes #4116)

Signed-off-by: Michael Eshom <oldiesmann@oldiesmann.us>

Sesquipedalian commented on Sources/ManageNews.php:
```diff
@@ -1008,7 +1008,7 @@ function SendMailing($clean_only = false)
 				array(
 					$row['email_address'],
-					!empty($_POST['send_html']) ? '<a href="' . $scripturl . '?action=profile;u=' . $row['id_member'] . '">' . $cleanMemberName . '</a>' : ($context['send_pm'] ? '[url=' . $scripturl . '?action=profile;u=' . $row['id_member'] . ']' . $cleanMemberName . '[/url]' : $cleanMemberName),
+					!empty($_POST['send_html']) ? '<a href="' . $scripturl . '?action=profile;u=' . $row['id_member'] . '">' . $cleanMemberName . '</a>' : ($context['send_pm'] ? '[url=' . $scripturl . '?action=profile;u=' . $row['id_member'] . ']' . $cleanMemberName . '[/url]' : $scripturl . '?action=profile;u=' . $row['id_member']),
```
How about `$cleanMemberName . ' (' . $scripturl . '?action=profile;u=' . $row['id_member'] . ')'`?

Oldiesmann replied:
We already have {$member.name} for the member name though, so I'm not sure why we need to include it with the member link when we're not sending HTML emails

Sesquipedalian replied:
As far as I can tell, the idea of `{$member.link}` seems to be to give a clickable version of the member's name that links to the member's profile. I figured that writing the name followed by the URL to the profile was the best way to approximate that in a plain text email.

Maybe we should have both a `{$member.link}` variable and a `{$member.url}` variable? Then `{$member.link}` would give the clickable name in HTML emails and PMs or the nearest approximation we can provide in plain text emails. Meanwhile, `{$member.url}` would provide just the raw `$scripturl . '?action=profile;u=' . $row['id_member']` string so that admins could explicitly include that even in HTML emails whenever they chose.

Oldiesmann replied:
The helptext for `{$member.link}` simply says "The current member's link". Most people would expect that to be a link to the current member's profile, so I'm not sure adding a `{$member.url}` would be very useful to people. On a related note we're doing the same thing with `{$latest_member.link}`.

Sesquipedalian replied:
Fair enough. :)

Sesquipedalian commented:
I suppose we should probably also give the same treatment to `$cleanLatestMember` a bit [earlier](https://github.com/SimpleMachines/SMF2.1/blob/release-2.1/Sources/ManageNews.php#L864) in the function.

Oldiesmann committed:
Merge branch 'release-2.1' of github.com:SimpleMachines/SMF2.1 into release-2.1

Oldiesmann committed:
Bad logic lead to newsletters not getting sent in most cases

Signed-off-by: Michael Eshom <oldiesmann@oldiesmann.us>

Oldiesmann commented:
Found the issue with newsletters not being sent for me (at least one of them - my non-admin test account is still not getting them) so I changed the name and comments for this PR...

Oldiesmann commented:
Still looking into the other newsletter issue here... If I change my regular test account to a global mod account it gets found by the query to get members to send emails to. I'll have to play with it more tomorrow and see what I can come up with.

Oldiesmann committed:
Fixes the two remaining issues:
- Always show a link for {$latest_member.link} as well
- Ignore $_POST['exclude_groups'] if it's an empty string - we don't want to exclude "Regular Members" by default.

Signed-off-by: Michael Eshom <oldiesmann@oldiesmann.us>
